### PR TITLE
Erin/edit casenote richtext

### PR DESCRIPTION
**Pull request by erinysong**
## FEATURE TITLE

Enabled rich text to casenote cards
Created Edit feature on casenote cards

### Related PRs

Optional - any related PRs you're waiting on, or PRs that will conflict, etc

### Migrations

Optional - if you added anything to the database through migration(s)

### Tests Performed, Edge Cases

Lol we'll see if we test

### Screenshots

Add screenshots!!! If you'd like

CC: @adowski

### CODE REVIEW

File: app/javascript/components/CaseNoteForm.jsx [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+            >
+              {this.state.type === "create" ? 'Submit Case Note' : 'Edit Casenote'}
+            </Button>
+          </DialogActions>
+        </Dialog>
+      );
+    }
+    else if (this.state.type === "delete") {
+      dialog = (
+        <Dialog
```
Comment (adowski) [inline app/javascript/components/CaseNoteForm.jsx]: Let's create a `DeleteModal` (or something similar) that handles this delete pop-up for us, so we can reuse it in the future! Will send more info

File: app/javascript/components/CaseNoteForm.jsx [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+        title: this.state.title,
+        description: this.state.description,
+        internal: this.state.internal,
+        participant_id: this.props.participantId,
+      };
+      apiPost('/api/case_notes', { case_note: body })
+        .then(() => window.location.reload())
+        .catch(error => console.error(error));
+    } else if (this.state.type === "edit") {
+      const newTitle = this.state.tempTitle;
```
Comment (adowski) [inline app/javascript/components/CaseNoteForm.jsx]: I don't think these temporary variables are necessary (on cancel, you can just clear the state to it's original content)

File: app/javascript/components/CaseNoteForm.jsx [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+    return (
+      <>
+        {this.button()}
+        {dialog}
+      </>
+    );
+  }
+}
+
```
Comment (adowski) [inline app/javascript/components/CaseNoteForm.jsx]: Please add in propTypes!

File: app/javascript/components/CaseNoteForm.jsx [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+        "title": this.state.tempTitle,
+        "description": this.state.tempDescription,
+        "internal": this.state.tempInternal,
+        "participant_id": this.state.participant_id,
+      };
+
+      body = JSON.stringify({case_note: body});
+      let req = '/api/case_notes/' + this.state.id;
+
+      fetch(req, {
```
Comment (adowski) [inline app/javascript/components/CaseNoteForm.jsx]: Can make this an `apiPatch()` call

File: app/javascript/components/CaseNoteForm.jsx [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+  handleDescriptionChange = name => state => {
+    // TODO: the line below is the rtf representation. Update to this once rtf on /casenotes
+    const value = JSON.stringify(convertToRaw(state.getCurrentContent()));
+    //const value = state.getCurrentContent().getPlainText();
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit() {
+    if (this.state.type === "create") {
+      const body = {
```
Comment (adowski) [inline app/javascript/components/CaseNoteForm.jsx]: You can set this body variable outside of the if statements so you don't have to repeat the same code for each `this.state.type` !

File: app/javascript/components/CaseNoteCard.jsx [hunk 6]
```diff
@@ -24,16 +34,106 @@ const styles = {
...
+    },
+    saveDocumentButtonStyle: {
+    borderStyle: 'solid 3px grey'
+    }
+}
+
+const defaultTheme = createMuiTheme();
+
+
+class SimpleMenu extends React.Component {
```
Comment (adowski) [inline app/javascript/components/CaseNoteCard.jsx]: Will push a commit to get rid of this second class

File: app/javascript/components/CaseNoteForm.jsx (hunk 1 above)
Comment (adowski) [reply app/javascript/components/CaseNoteForm.jsx]: Also, you can render the `DeleteModal` directly in the `CaseNoteCard` so that the form doesn't get too messy!

image
